### ISSUE TITLE
Option to run Flycheck after buffer switch

### ISSUE THREAD

**Issue by raxod502** [flycheck/flycheck#1297]
I'd like to have Flycheck run after a short delay (specifically, after `flycheck-idle-change-delay`) whenever I switch buffers.

The current behavior is that syntax checks occur only after modifications to the buffer, or after a save, as per the default setting of `flycheck-check-syntax-automatically`.

However, it often happens that I change some configuration externally, or I make modifications in another file. Then, when I switch back to an old buffer that is supposed to now have errors in it, Flycheck does not run another check. I have to manually edit the buffer or save it. My current workflow is to switch to a buffer, insert a character, delete it, and save—after that, Flycheck will show the errors. And I feel really dumb every time I do that.

I'd propose for a new value `buffer-change` to be added to `flycheck-check-syntax-automatically`, which would cause a syntax check to be run after `flycheck-idle-change-delay` (or an analogous variable), whenever the user switched buffers.

**Comment by fmdkdd** [comment]
Thanks for the idea.  I like it!

The only downside I'm seeing is that with this new setting, a syntax check would begin every time you switch to this buffer, even if the buffer hasn't changed, right?

But what if your configuration file hasn't changed either?  You don't need to run the syntax check, but Flycheck cannot know because it has no knowledge of the dependencies between your buffers.

It looks like the correct solution here is to make Flycheck *project-aware*, which is out of scope for now.

However, your suggestion is a nice compromise to that.  We don't *need* to know the dependencies, we just run the syntax check regardless.  

I'd be wary about making this new option part of the default value for `flycheck-check-syntax-automatically`, but otherwise I'm okay offering it as a choice.

**Comment by raxod502** [comment]
I agree with all the things you have said. I might see if I can make a pull request (but that depends on how complicated it turns out to be).

**Comment by fmdkdd** [comment]
@raxod502 Sure, that would be lovely.  You can make an incomplete PR and we can answer any questions you might have in there directly.

**Comment by raxod502** [comment]
I have created pull request #1308 to implement this feature.
